test/Cli/Func.E2E.Tests/Commands/FuncInit/NodeInitTests.cs

```diff
@@ -0,0 +1,204 @@
+{
+    [Trait(WorkerRuntimeTraits.WorkerRuntime, WorkerRuntimeTraits.Node)]
+    public class NodeInitTests(ITestOutputHelper log) : BaseE2ETests(log)
+    {
+        [Theory]
+        [InlineData("")]
+        [InlineData("v3")]
+        [InlineData("v4")]
+        public void Init_WithSupportedModel_SuccessfulExecution(string programmingModel)
+        {
+            var workingDir = WorkingDirectory;
+            var programmingModelFlag = string.IsNullOrEmpty(programmingModel) ? string.Empty : $"--model {programmingModel}";
+            var testName = nameof(Init_WithSupportedModel_SuccessfulExecution);
+            var funcInitCommand = new FuncInitCommand(FuncPath, testName, Log ?? throw new ArgumentNullException(nameof(Log)));
+            var localSettingsPath = Path.Combine(workingDir, Common.Constants.LocalSettingsJsonFileName);
+            var expectedcontent = new[] { Common.Constants.FunctionsWorkerRuntime, "node" };
+            var packageJsonPath = Path.Combine(workingDir, Common.Constants.PackageJsonFileName);
+            var expectedJsoncontent = new[] { $"\"@azure/functions\": \"^4" };
+            var filesToValidate = new List<(string FilePath, string[] ExpectedContent)>
+            {
+                (localSettingsPath, expectedcontent)
+            };
+            if (programmingModel == "v4" || string.IsNullOrEmpty(programmingModel))
+            {
+                filesToValidate.Add((packageJsonPath, expectedJsoncontent));
+            }
+
+            // Initialize node function app
+            var funcInitResult = funcInitCommand
+               .WithWorkingDirectory(workingDir)
+               .Execute(["--worker-runtime", "node", programmingModelFlag]);
+
+            // Validate expected output content
+            funcInitResult.Should().WriteVsCodeExtensionsJsonAndExitWithZero(workingDir);
+            funcInitResult.Should().FilesExistsWithExpectContent(filesToValidate);
+        }
+
+        [Theory]
+        [InlineData("v1")]
+        [InlineData("v2")]
+        public void Init_WithUnsupportedModel_FailsWithError(string programmingModel)
+        {
+            var workingDir = WorkingDirectory;
+            var testName = nameof(Init_WithUnsupportedModel_FailsWithError);
+            var funcInitCommand = new FuncInitCommand(FuncPath, testName, Log ?? throw new ArgumentNullException(nameof(Log)));
+
+            // Initialize node function app
+            var funcInitResult = funcInitCommand
+                .WithWorkingDirectory(workingDir)
+                .Execute(["--worker-runtime", "node", "--model", programmingModel]);
+
+            // Validate expected output content
+            funcInitResult.Should().ExitWith(1);
+            funcInitResult.Should().HaveStdErrContaining($"The {programmingModel} programming model is not supported for worker runtime node.");
+        }
+
+        [Fact]
+        public void Init_VerifyHostJsonFileExpectsLoggingConfig_SuccessfulExecution()
```

Comment:
Instead of using "SuccessfulExecution" in all of these tests, can we try to describe expected result? I.e.

```suggestion
        public void Init_GeneratesHostJson_ContainsExpectedLoggingConfig()
```

Reply:
This feedback applies to every test you have written "SuccessfulExecution"